packages/network/src/network.js

```diff
@@ -108,12 +108,12 @@ const prepareHalfConnection = (zone, { when, watch }, makeAckWatcher) => {
           throw closed;
         }
 
-        return watch(
-          E(handlers[r])
-            .onReceive(conns.get(r), toBytes(packetBytes), handlers[r])
-            .catch(rethrowUnlessMissing),
```

Comment:
oh... this catch went away? is that on purpose?

Reply:
Yes, because `watch` only works correctly for surviving upgrade trauma when it is handed a remotely-decided promise.

Reply:
As @iomekam and I discussed earlier, this `watch` with catch needs to become two `watch`es, like:

```js
  const innerVow = watch(E(handlers[r]).onReceive(conns.get(r), toBytes(packetBytes), handlers[r]));
  return watch(innerVow, rethrowUnlessMissingWatcher);
```

Reply:
Maybe I'd phrase it as "`catch` only works correctly if it isn't used on a remotely-decided promise".